dicomservice/compiler-plugin/src/main/java/io/ballerinax/health/dicom/dicomservice/compiler/DicomDiagnosticCode.java

```diff
@@ -0,0 +1,42 @@
+package io.ballerinax.health.dicom.dicomservice.compiler;
+
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+
+/**
+ * DICOM diagnostic codes.
+ */
+public enum DicomDiagnosticCode {
+    DICOM_101("DICOM_101", "remote methods are not allowed in dicomservice:Service", DiagnosticSeverity.ERROR),
+    DICOM_102("DICOM_102", "invalid number of parameters", DiagnosticSeverity.ERROR),
+    DICOM_103("DICOM_103",
+            "invalid resource method first parameter type: expected \"" + Constants.ALLOWED_RESOURCE_FIRST_PARAM
+                    + "\", but found \"%s\"", DiagnosticSeverity.ERROR),
```

Comment:
Use string templates?

Reply:
String templates were introduced in Java 21. However, we are using Java 17. So used `String.format()` instead in https://github.com/ballerina-platform/module-ballerinax-health.dicom/commit/3807383514d097fe372c936aa46ef84a627a58d3